torch/csrc/api/include/torch/nn/modules/container/moduledict.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <torch/nn/cloneable.h>
+#include <torch/nn/module.h>
+#include <torch/ordered_dict.h>
+
+#include <vector>
+
+namespace torch {
+namespace nn {
+
+class ModuleDictImpl : public Cloneable<ModuleDictImpl> {
+ public:
+  using Iterator = OrderedDict<std::string, std::shared_ptr<Module>>::Iterator;
+  using ConstIterator =
+      OrderedDict<std::string, std::shared_ptr<Module>>::ConstIterator;
+
+  ModuleDictImpl() = default;
+
```

Comment:
It would be awesome to add a `ModuleDictImpl` constructor that takes a `torch::OrderedDict` as input, so that we can do something similar to:
```cpp
torch::OrderedDict<torch::nn::Module> modules = {{"a", ModuleA()}, {"b", ModuleB()}};
auto moduledict = torch::nn::ModuleDict(modules);
```